src/modules/fw_wind_estimator/FixedwingWindEstimator.cpp

```diff
@@ -0,0 +1,300 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "FixedwingWindEstimator.hpp"
+
+using namespace time_literals;
+using namespace matrix;
+
+using math::constrain;
+using math::interpolate;
+using math::radians;
+
+FixedwingWindEstimator::FixedwingWindEstimator(bool vtol) :
+	ModuleParams(nullptr),
+	ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::nav_and_controllers),
+	_loop_perf(perf_alloc(PC_ELAPSED, MODULE_NAME": cycle"))
+{
+	/* fetch initial parameter values */
+	parameters_update();
+
+}
+
+FixedwingWindEstimator::~FixedwingWindEstimator()
+{
+	perf_free(_loop_perf);
+}
+
+bool
+FixedwingWindEstimator::init()
+{
+	if (!_vehicle_angular_velocity_sub.registerCallback()) {
+		PX4_ERR("callback registration failed");
+		return false;
+	}
+
+	return true;
+}
+
+int
+FixedwingWindEstimator::parameters_update()
+{
+
+	_mass = _param_fw_w_mass.get();
+	_stall_airspeed = _param_fw_airspd_stall.get();
+
+	return PX4_OK;
+}
+
+void
+FixedwingWindEstimator::vehicle_land_detected_poll()
+{
+	if (_vehicle_land_detected_sub.updated()) {
+		vehicle_land_detected_s vehicle_land_detected {};
+
+		if (_vehicle_land_detected_sub.copy(&vehicle_land_detected)) {
+			_landed = vehicle_land_detected.landed;
+		}
+	}
+}
+
+void FixedwingWindEstimator::airspeed_poll()
+{
+	airspeed_validated_s airspeed_validated;
+
+	if (_airspeed_validated_sub.update(&airspeed_validated)) {
+		_calibrated_airspeed = math::max(0.5f, airspeed_validated.calibrated_airspeed_m_s);
+		_true_airspeed = math::max(0.5f, airspeed_validated.true_airspeed_m_s);
+	}
+}
+
+
+void
+FixedwingWindEstimator::vehicle_attitude_poll()
+{
+	if (_vehicle_attitude_sub.update(&_vehicle_attitude)) {
```

Comment:
```suggestion
	vehicle_attitude_s vehicle_attitude;
	if (_vehicle_attitude_sub.update(&vehicle_attitude)) {
```
no need to save the whole message. Same for the other messages